cla_backend/apps/cla_butler/qs_to_file.py

```diff
@@ -0,0 +1,134 @@
+import re
+import time
+
+from django.db import IntegrityError
+from django.db.models import ForeignKey
+from django.utils.dateparse import parse_datetime
+
+from jsonfield import JSONField
+
+from cla_common.money_interval.fields import MoneyIntervalField
+from cla_common.money_interval.models import MoneyInterval
+
+
+logger = logging.getLogger(__name__)
+
+
+RE_DATE = re.compile(r'(\d{4})-(\d{2})-(\d{2})$')
+RE_DATETIME = re.compile(r'(\d{4})-(\d{2})-(\d{2}) (\d{2}):(\d{2}):(\d{2})\.(\d{6})\+(\d{2}):(\d{2})$')
+WRITE_MODE = 'wb'
+APPEND_MODE = 'a'
+
+
+class QuerysetToFile(object):
+    def __init__(self, path):
+        if not os.path.exists(path):
+            os.makedirs(path)
+        self.path = path
+
+    def get_file_path(self, model, ext='csv'):
+        return os.path.join(self.path, '%s.%s' % (model.__name__, ext))
+
+    def get_name(self, field):
+        field_name = field.name
+        if isinstance(field, ForeignKey):
+            field_name = '%s_id' % field_name
+        return field_name
+
+    def get_value(self, instance, field):
+        val = getattr(instance, self.get_name(field))
+        if isinstance(val, MoneyInterval):
+            val = json.dumps(val.as_dict())
+        if hasattr(val, 'pk'):
+            val = val.pk
+        if val is None:
+            val = ''
+        try:
+            return unicode(val).encode('utf-8')
+        except UnicodeDecodeError:
+            return val
+
+    def dump_to_csv(self, qs):
+        """dump queryset to .csv"""
+        file_path = self.get_file_path(qs.model)
+
+        if os.path.isfile(file_path):
+            write_mode = APPEND_MODE
+        else:
+            write_mode = WRITE_MODE
+
+        field_names = [self.get_name(f) for f in qs.model._meta.fields]
+        with open(file_path, write_mode) as csvfile:
+            writer = csv.writer(csvfile, quoting=csv.QUOTE_ALL)
+            if write_mode == WRITE_MODE:
+                writer.writerow(field_names)
+            for instance in qs.iterator():
+                writer.writerow(
+                    [self.get_value(instance, f) for f in
+                     qs.model._meta.fields])
+            csvfile.close()
+
+    def dump(self, qs):
+        logger.info('starting dump of %s' % qs.model.__name__)
+        start = time.time()
+        self.dump_to_csv(qs)
+        logger.info('Time to dump %s: %s' % (qs.model.__name__, time.time() - start))
+
+    def set_value(self, val, field):
+        if val == '' and field.empty_strings_allowed and not field.null:
+            val = ''
+        elif val in field.empty_values:
+            val = None
+        elif isinstance(field, MoneyIntervalField):
+            val = MoneyInterval.from_dict(json.loads(val))
+        elif isinstance(field, JSONField):
+            val = json.dumps(val)
+        elif RE_DATE.match(val):
+            val = datetime.strptime(val, '%Y-%m-%d').date()
+        elif RE_DATETIME.match(val):
+            val = parse_datetime(val)
```

Comment:
Are regexes the best way to do this? Would [dateutil](https://dateutil.readthedocs.io/en/stable/parser.html) help here?